rust-wasm-id-allocator/distributed-id-allocator/src/compressor.rs

```diff
@@ -232,8 +233,16 @@ impl IdCompressor {
         }
     }
 
+    #[cfg(feature = "uuid-generation")]
     pub fn deserialize(bytes: &[u8]) -> Result<IdCompressor, DeserializationError> {
-        persistence::deserialize(bytes)
+        persistence::deserialize(bytes, || SessionId::new())
+    }
+
+    pub fn deserialize_with_session_id(
+        bytes: &[u8],
+        session_id: SessionId,
+    ) -> Result<IdCompressor, DeserializationError> {
+        persistence::deserialize(bytes, || session_id)
```

Comment:
Plumb delegate pattern through to wasm layer

Reply:
Done!